test/src/commonMain/kotlin/io/realm/util/Utils.kt

```diff
@@ -32,13 +36,28 @@ expect object PlatformUtils {
 
 // Platform independent helper methods
 object Utils {
+
     fun createRandomString(length: Int): String {
         val allowedChars = ('A'..'Z') + ('a'..'z') + ('0'..'9')
         return (1..length)
             .map { allowedChars.random() }
             .joinToString("")
     }
+
     fun printlntid(message: String) {
         println("<" + PlatformUtils.threadId() + "> $message")
     }
 }
+
+/**
+ * Helper method for easily updating a single object. The updated object will be returned.
+ * This method control its own write transaction, so cannot be called inside a write transaction
+ */
+suspend fun <T : RealmObject> T.update(block: T.() -> Unit): T {
+    val realm = ((this as RealmObjectInternal).`$realm$Owner` as RealmReference).owner as Realm
+    return realm.write {
+        val liveObject: T = findLatest(this@update)!!
```

Comment:
This was mostly for testing where it would be a mistake if the object was deleted. But I'll move the method to the public API and change it a little bit.

Reply:
Ok. I guess we can discuss if we consider to add a similar method in the actual library.